### DISCUSSION TITLE
File Menu Event + Search Panel = Amazing new Workflow

### DISCUSSION THREAD

**Discussion by FelipeRearden** [Taitava/obsidian-shellcommands#177] [Show and tell]
Hello @Taitava !!!!!!

Just to share an amazing workflow that I discovered today (if I did not write this I couldn't sleep today)

File Menu Events can trigger SC commands from the SEARCH PANE !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
![50](https://user-images.githubusercontent.com/85180696/157102968-653bb0a0-0fa6-4006-bfd8-3b5aa72c8a8c.png)

This way I can write an SC that place a link of the file that I´m searching inside my active note on caret position!!!!!!

`echo [[{{event_filename}}]]` output=current file caret position event=file menu.

The sky is the limit, I can create an SC that will open the search file in the end for example ....

`echo {{event_file_name}}:new-pane:-1:-1` output=open a file event=file menu

this is amazing !!!!! 

have a great day!!!!!!!!

**Comment by Taitava** [comment]
Really nice! 😎 

I did not know that the file menu event occurs in the search results, too. I did notice some time ago that if you right click a link in a note, then the file menu event shell commands appear there.

I'll add a short mention of your finding in the documentation here: https://publish.obsidian.md/shellcommands/Events/File+menu

Have a great day, too! 🙂👍
